Orb improvements [semver:patch]

srijan-deepsource opened the pull request.

srijan-deepsource committed:
orb improvements

KyleTryon commented on src/@orb.yml:
```diff
@@ -0,0 +1,7 @@
+version: 2.1
+description: >
+  CircleCi orb that uploads test coverage reports to DeepSource for Test Coverage analysis.
```
CC @nickbenCCI for review

KyleTryon commented on .circleci/config.yml:
```diff
@@ -22,6 +22,8 @@ workflows:
           requires: [orb-tools/lint]
       - orb-tools/publish-dev:
           orb-name: deepsource/test-coverage
+          publish-sha-version: false
```
The sha version is used for the final publishing step.

KyleTryon replied:
actually, this line can just be removed.

srijan-deepsource replied:
Sure. I was using `orb-tools@9.0.0` in the previous version and noticed that the sha-version was being published instead of `dev:alpha`.

I'll make the changes here. Waiting for the complete review.

KyleTryon commented:
@srijan-deepsource Could you add `[semver:patch]` to the title? if we are setup properly, this will automatically issue the patch update when merged.

It is important that the semver tag makes it into the final commit message (squash merge normally does this automatically).

srijan-deepsource commented:
> @srijan-deepsource Could you add `[semver:patch]` to the title? if we are setup properly, this will automatically issue the patch update when merged.
> 
> It is important that the semver tag makes it into the final commit message (squash merge normally does this automatically).

Done. I guess I'll need to set up a promote-dev step for this?

KyleTryon commented:
> > @srijan-deepsource Could you add `[semver:patch]` to the title? if we are setup properly, this will automatically issue the patch update when merged.
> > It is important that the semver tag makes it into the final commit message (squash merge normally does this automatically).
> 
> Done. I guess I'll need to set up a promote-dev step for this?

Hello @srijan-deepsource I did not realize before that you are not using our pre-configured pipeline generated by the orb-starter-kit (or released as of yesterday, the new orb development kit). 

If you are manually publishing your orb there is no need to adjust the title. If you would like to setup a fully automated CI pipeline, I would love to assist you with that. It makes iterating changes quite a bit easier in the future. 

You can see an example of that and some documentation on it here: https://github.com/CircleCI-Public/Orb-Project-Template/tree/master/.circleci

Reconfirming with @nickbenCCI for review of description

nickbenCCI commented:
Updated orb description:
Upload test coverage reports to DeepSource for application performance analysis. Developers can identify and fix bug risks and security flaws pre-production.

srijan-deepsource committed:
Update description and add step to publish orb

srijan-deepsource commented:
@nickbenCCI @KyleTryon : Updated the description. Also added the step to promote orb to production.

I have this question: since the orb follows the "unpacked" design now, how can I validate the orb locally?
I tried `circleci orb validate src/@orb.yml` and it always says that the orb is `valid` even when the commands had an error.
What am I missing here?

KyleTryon commented:
> @nickbenCCI @KyleTryon : Updated the description. Also added the step to promote orb to production.
> 
> I have this question: since the orb follows the "unpacked" design now, how can I validate the orb locally?
> I tried `circleci orb validate src/@orb.yml` and it always says that the orb is `valid` even when the commands had an error.
> What am I missing here?

Hello @srijan-deepsource,
The orb can be packed and validated locally if desired (though we typically recommend allowing the automatic pipeline to handle this. https://circleci.com/docs/2.0/creating-orbs/#section=configuration)

You would use the orb pack command:

```
circleci orb pack <src> > orb.yml
```

and then validate as you mention above.

For updating the orb, if you are not using the Orb Development Kit pipeline (https://circleci.com/docs/2.0/creating-orbs/#issue-a-new-release), you can then also manually publish the orb as well from the CLI

```
circleci orb publish orb.yml namespace/orb@version
```

srijan-deepsource commented:
> Hello @srijan-deepsource,
> The orb can be packed and validated locally if desired (though we typically recommend allowing the automatic pipeline to handle this. https://circleci.com/docs/2.0/creating-orbs/#section=configuration)
> 
> You would use the orb pack command:
> 
> ```
> circleci orb pack <src> > orb.yml
> ```
> 
> and then validate as you mention above.
> 
> For updating the orb, if you are not using the Orb Development Kit pipeline (https://circleci.com/docs/2.0/creating-orbs/#issue-a-new-release), you can then also manually publish the orb as well from the CLI
> 
> ```
> circleci orb publish orb.yml namespace/orb@version
> ```

Got it! Thanks. I have included a step to publish the prod version in the PR. Waiting for a review from your team.

KyleTryon commented:
Hello 👋 

`orb-tools/publish-dev`, and `orb-tools/dev-promote-prod-from-commit-subject` will both need access to a context which contains the CIRCLE_TOKEN environment variable (with your API key) in order to publish the dev and production versions of the orb.

Example: https://github.com/CircleCI-Public/shellcheck-orb/blob/d30bb18f60d683275f16316fb48ccafeed6f6464/.circleci/config.yml#L69

srijan-deepsource committed:
set orb ref to promote the "dev:alpha" version

srijan-deepsource commented:
> Hello 👋
> 
> `orb-tools/publish-dev`, and `orb-tools/dev-promote-prod-from-commit-subject` will both need access to a context which contains the CIRCLE_TOKEN environment variable (with your API key) in order to publish the dev and production versions of the orb.
> 
> Example: https://github.com/CircleCI-Public/shellcheck-orb/blob/d30bb18f60d683275f16316fb48ccafeed6f6464/.circleci/config.yml#L69

Hey @KyleTryon! We already have an environment variable `CIRCLE_TOKEN` containing the API key. I guess the `context` defaults to the same env variable if not provided explicitly?

Also, am having no trouble publishing the dev version (see the passed checks).
Please let me know if there's anything else?

KyleTryon commented:
@srijan-deepsource That is perfectly fine then 👍 Lets merge this in
